### PR TITLE
InfluxDB: Upgrade InfluxDB in devenv

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Upgrade from InfluxDB 1.7 to 1.8 in devenv

I'm also doing some serious cleanup of the Flux code, mostly de-exporting symbols not used outside of the package (the less exports the better).

### CODE REVIEW

**Comment by papagian** [comment]
For `mysql` and `postgres` we support [dynamically changing](https://github.com/grafana/grafana/pull/18112) the docker image. We could do the same for InfluxDB.

Also I [suspect](https://github.com/grafana/grafana/blob/master/devenv/datasources.yaml#L30) that we shouldn't have had `/api/v2` suffix even before.

**Comment by aknuds1** [comment]
I can look into making the image configurable @papagian. About the suffix, it is actually necessary before the upgrade, I tested.

**Comment by aknuds1** [comment]
@papagian Could you show me exactly how to make the InfluxDB tag configurable in devenv? I found that MySQL uses the variable mysql_version, but when I grep for it, I can only find where it's used.

**Comment by papagian** [comment]
> @papagian Could you show me exactly how to make the InfluxDB tag configurable in devenv? I found that MySQL uses the variable mysql_version, but when I grep for it, I can only find where it's used.

You only need to create an `.env` file under influxdb block with an `influxdb_version` variable and change the image in the docker-compose file to use this variable. The you can use the command:
`make devenv sources=influxdb influxdb_version=1.8.1-alpine` because values in the shell take precedence over those specified in the .env file.

File: devenv/docker/blocks/influxdb/docker-compose.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
   influxdb:
-    image: influxdb:1.7.6
+    image: influxdb:1.8.1-alpine
```
Comment (ryantxu) [inline devenv/docker/blocks/influxdb/docker-compose.yaml:2]: Am i right that this is the only thing related to the issue title?  The rest is making things private

**Review by ryantxu — APPROVED** [review]
seems fine -- the exported stuff is like that because it came from a package where it is appropriate to be exported... now in core it should not be.

However I do think we should put all the `XxxToOptionalYyy` helpers in the SDK -- they are currently copied in most datasources

File: devenv/docker/blocks/influxdb/docker-compose.yaml (hunk 1 above)
Comment (aknuds1) [reply devenv/docker/blocks/influxdb/docker-compose.yaml:2]: You're right @ryantxu. Originally, I was also upgrading the InfluxDB Go client, but I ended up doing that in another PR to fix a serious bug of leaking TCP connections.

While I was working on the code, I figured I could also tighten up the code.

**Pull request merged**